File: src/main/resources/sum_of_digits.rb
puts "Enter a number to calculate the sum of its digits:"
number = gets.chomp.to_i

sum = 0

# Calculate the sum of the digits using a loop
while number > 0
  digit = number % 10
  sum = sum + digit
  number = number / 10
end

# Print the sum of the digits
puts "The sum of the digits is: #{sum}"

File: src/main/resources/SumOfDigits.kt
fun main() {
    print("Enter a number to calculate the sum of its digits: ")
    var number = readln().toInt()

    var sum = 0

    while (number > 0) {
        val digit = number % 10
        sum += digit
        number /= 10
    }

    println("The sum of the digits is: $sum")
}